packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1781,7 +1781,6 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
       final RenderObject parent = this.parent;
       relayoutBoundary = parent._relayoutBoundary;
     }
-    assert(parent == this.parent);
```

Comment:
It was supposed to be testing the "parent" from the block just before this line. It's fine to remove.